inloop/grading/admin.py

```diff
@@ -1,6 +1,66 @@
+
 from django.contrib import admin
+from django.http import HttpResponse
 
 from inloop.grading.models import DetectedPlagiarism, PlagiarismTest
 
-admin.site.register(DetectedPlagiarism)
-admin.site.register(PlagiarismTest)
+
+def download_jplag_zip(modeladmin, request, queryset):
+    """Fetch the stored jplag zip file and return it as a download"""
+    if queryset.count() != 1:
+        modeladmin.message_user(
+            request, "You can not download more or less than one zip file at a time."
+        )
+        return
+    test = queryset[0]
+    if not test.zip_file:
+        modeladmin.message_user(
+            request, "The selected plagiarism test has no downloadable zip file."
+        )
+    response = HttpResponse(open(test.zip_file.path, 'rb'), content_type='application/zip')
+    response[
+        'Content-Disposition'
+    ] = 'attachment; filename=' + 'JPlag_Output_' + str(test.created_at).replace(' ', '_') + '.zip'
```

Comment:
Style: you should use format strings instead.

Reply:
Fixed.